src/catnip/runtime/memory/manager.rs

```diff
@@ -86,17 +87,14 @@ impl MemoryManager {
     pub fn into_sgarray(&self, buf: DPDKBuf) -> Result<dmtr_sgarray_t, Fail> {
         let (mbuf_ptr, sgaseg): (*mut rte_mbuf, dmtr_sgaseg_t) = match buf {
             // Heap-managed buffer.
-            DPDKBuf::External(bytes) => {
-                // We have to do a copy here since `Bytes` uses an `Arc<[u8]>` internally and has
-                // some additional bookkeeping for an offset and length, but we want to be able to
-                // hand off a raw pointer up the application that they can free later.
```

Comment:
Are we changing the contract with the user here?  This comment makes it sound like the user was allowed to call "free" (i.e. the default memory allocators free() routine) on any of the sga_seg_buf pointers.  Would the new code still allow this?  Or must the user now call free_sgarray()?  I'm okay with the latter, just wondering what we're changing here.

We really need a document that defines the application level interface precisely.

Reply:
I don't know who wrote this. This comment was outdated and wrong, that is why I dropped it. The user is not allowed to call free when using memory allocated this way. We don't support use-after free. For this we should provider full integration with C's memory allocator, which is on the roadmap.

Reply:
Okay, thanks.